pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -607,8 +625,64 @@ func (c *Controller) getNodePortGatewayServices() []*model.Service {
 	return out
 }
 
-// updateServiceExternalAddr updates ClusterExternalAddresses for ingress gateway service of nodePort type
-func (c *Controller) updateServiceExternalAddr(svcs ...*model.Service) bool {
+func (c *Controller) updateNetworkGateways(svc *model.Service) {
+	c.Lock()
+	defer c.Unlock()
+	svc.Mutex.RLock()
```

Comment:
nit: should we move this into extractGatewaysFromService? I think we only need it there

Reply:
I merged the two and moved it all to the network.go file